### PR TITLE
Fixed Arc Clipping Issue

### PR DESCRIPTION
**Pull request by schwindy**
# Overview
I added configurable Globe Arc constants that change Arc Generation Behavior. In general, these prevent Arcs from clipping into the Earth Sphere with the current values.

The following Arc constants were added:
1. `ARC_DISTANCE_OFFSET` - Arc Point Distance at which Arc Length should begin increasing linearly
2. `ARC_LENGTH_OFFSET` - Minimum Arc Length to be added to Calculated Arc Length (based on distance)
3. `ARC_MAX_DISTANCE` - Maximum Distance Allowed between Two Arc Points in order to draw an Arc

`ARC_DISTANCE_OFFSET` adds extra length, linearly scaling with distance > ARC_DISTANCE_OFFSET, to Arcs where the distance between the Two Arc Points is greater than ARC_DISTANCE_OFFSET.

`ARC_LENGTH_OFFSET` adds extra length to all Arcs, increasing the height of the Arc Mid-Point, which increases the "height" (above globe) the Arc uses.

`ARC_MAX_DISTANCE` can prevent curves from clipping into the Earth Sphere, but alone and if set to a value lower than the distance between the largest Arcs will also prevent those Arcs with exceptionally large distance from ever being drawn if the distance is > this maximum value.

## Testing Steps
- [ ] Ensure Arcs NEVER clip into the Earth Sphere
  - [ ] Rotate Earth in as many ways as possible to attempt to draw ALL Arcs
  - [ ] All Arcs (red lines) should be above the Earth Geometry at all times
  - [ ] Ensure "Small" Arc Angle/Distance is desired (not too high, never clips, etc.)
  - [ ] Ensure "Medium" Arc Angle/Distance is desired (not too high, never clips, etc.)
  - [ ] Ensure "Large" Arc Angle/Distance is desired (not too high, never clips, etc.)

### CODE REVIEW

File: src/constants/index.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
 /* eslint-disable import/prefer-default-export */
-export const TOTAL_ARCS = 20;
-
 export const COLOR_1 = '#ef0018';
 
+/**
+ * Maximum Length of Globe Curves that are drawn between Globe Points
```
Comment (schwindy) [inline src/constants/index.js:5]: I personally like verbose config doc blocks like this so that in the future it's clear what a given piece of config does.

However, let me know if this is "too" verbose or if you'd prefer no notes here. I can also add comments to all the other constants if you'd like 🐒

File: src/objects/drawCurve/index.js [hunk 2]
```diff
@@ -29,15 +29,17 @@ export default function drawCurve(a, b) {
 
   const lineGeometry = new THREE.BufferGeometry();
   const positions = new Float32Array(points.length * 3);
-  for (let ii = 0; ii < points.length; ii += 1) {
-    positions[ii * 3 + 0] = points[ii].x;
-    positions[ii * 3 + 1] = points[ii].y;
-    positions[ii * 3 + 2] = points[ii].z;
+  for (let i = 0; i < points.length; i += 1) {
```
Comment (schwindy) [inline src/objects/drawCurve/index.js:32]: I figured `i` was better than `ii` here, as there is no `i` used in this method 🤷 All changes to this file are completely semantics / formatting and do not change behavior in any way.

Happy to revert these changes if you'd prefer. 🐒

File: src/constants/index.js [hunk 3]
```diff
@@ -1,10 +1,35 @@
 /* eslint-disable import/prefer-default-export */
-export const TOTAL_ARCS = 20;
 
-export const COLOR_1 = '#ef0018';
+/**
+ * Arc Point Distance at which Arc Length should begin increasing linearly to account for Arcs with
+ * exceptionally large distance (ex: across globe, 180 degrees).
+ *
+ * @note This dynamically adds scaling extra height to Arc Mid-Points when the distance between the
```
Comment (elliottregan) [inline src/constants/index.js:7]: Nice. Thanks!

File: src/objects/drawCurve/index.js (hunk 2 above)
Comment (elliottregan) [reply src/objects/drawCurve/index.js:33]: Nah, that's fine. `ii` was from when this was all in one GIANT file.

**Review by trevormiles — APPROVED** [review]
